### DISCUSSION TITLE
Refresh data from useLazyAsyncData between dynamic pages

### DISCUSSION THREAD

**Discussion by Zibbp** [nuxt/nuxt#17645] [Questions] [answered]
Hi,

I have a dynamic page, `[name].vue` which fetches data from an API via the name from the router using useLazyAsyncData.
```
const {
  data: videos,
  pending,
  refresh,
} = await useLazyAsyncData("videos", () =>
  $fetch(
    `${config.apiURL}/videos/channel/${route.params.name}?limit=${paginationLimit.value}&page=${paginationPage.value}`
  )
);
```
This works great for first page I visit, in this case let's say I visit `/123`, the data for ID 123 is populated. Now if I use a NuxtLink to navigate back and visit another ID such as `/345` using a NuxtLink the data for `345` is not fetched, it's still using the data from `123`. 

Is there a way for the data to be refreshed when navigating to a new dynamic page?

**Comment by Megarsh** [comment, marked as answer]
You're using the same key (videos) for the dynamic pages that's why the data is pulled from the cache.

Try setting a unique key for any of those dynamic pages

```
const route = useRoute()

const {
  data: videos,
  pending,
  refresh,
} = await useLazyAsyncData(`videos-${route.params.name}`, () =>
  $fetch(
    `${config.apiURL}/videos/channel/${route.params.name}?limit=${paginationLimit.value}&page=${paginationPage.value}`
  )
);
```

**Reply by Zibbp** [reply]
Did not know that the key had to be unique. Modified it and it's working. Thanks!
